### DISCUSSION TITLE
How to prevent render component untill its ready

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3119] [Q&A]
I want to create component which has different element, like text,icon. So its take some time to load css and other things so that time its display white space with text and its look very ugly and once css load its look good. 
I figure out one solution is  to use like `web-button:not(:defined)` but I have to write it for all component manually,

Does there any way its atomically hide element untill its load fully .

**Comment by ynxtynxt** [comment]
Like using until directive? https://lit.dev/docs/templates/directives/#until
Define your own parent class or mixin to define the behavior?

**Comment by e111077** [comment]
How are you loading CSS? Lit's `static styles = css...` should handle styles loading at first render. You may run into issues if you use a `<link rel="stylesheet">` in your template because the browser will not start that load until the template is already rendered. You can read more on that here:

https://lit.dev/docs/components/styles/#external-stylesheet

**Comment by dbwodlf3** [comment]
I think this is very speical case.
I don't know exactly what is your problem, but just i see, I think you need to use custom event.
Just every element styles make as a "display:none" or something and just show process loading element.
and after your slow component is load css(first updated or something,), just change flag as loaded. and change display:none as display something.

like this.
```ts
@property({type: Boolean})
_load = false

static style = css`.none:{display:none}`

loadComponent(e){
    this._load = true;
}

render(){
  return  html`
    <div class="${this._load ? '' : 'none' }">
      <your-component1></your-component1>
      <your-component2></your-component2>
      <your-component3></your-component3>
      <your-component4></your-component4>
      <very-slow-component @load=${this.loadComponent}></very-slow-component>
    </div>
  `
}
```
